COBOL Programming Course #2 - Advanced Topics/COBOL Programming Course #2 - Advanced Topics.md

```diff
@@ -665,6 +665,333 @@ In a multithreaded environment, there are some limitations on COBOL programs. In
 To see more details on the limitation of COBOL with multithreading, check out the [Programming Guide](https://www.ibm.com/docs/en/cobol-zos/6.3?topic=multithreading-handling-cobol-limitations).
 
 \newpage
+
+# Subprograms
+when control is transferred from the active program to an external program, but the transferring program remains active
+and control can be returned to it,
+the program to which control is transferred is called a **subprogram**.
+
+This chapter discusses subprograms, the different between them and a paragraph or a normal program
+, and how to call and work with them
+
+- **Subprograms in COBOL**
+    - **Difference between a paragraph and a subprogram**
+    - **Different between a program and subprogram**
+
+- **Calling and Passing Arguments to a Subprogram**
+    - **The contents of The subprogram**
+    - **The subprogram Language**
+
+- **Flow of control between programs and subprograms**
+
+- **Rules for calling subprograms**
+    - **Location of subprogram**
+    - **Translation**
+    - **Compilation**
+    - **Link-editing**
+    - **CICS CSD entries without autoinstall program**
+    - **Recursive calls in COBOL**
+    - **Passing parameters to a subprogram**
+    - **Return from a subprogram**
+    - **Storage**
+    - **CICS condition, AID, and `abend` handling**
```

Comment:
Hi Ahmed, thanks for the PR! Would you mind expanding this chapter more? 

Check out Chapter 26 and 27 on this [Programming Guide](https://publibfp.boulder.ibm.com/epubs/pdf/igy6pg30.pdf). We are still missing some important concepts for subprograms (example includes, but not limited to, LINKAGE SECTION, ending programs - effects of STOP RUN, GOBACK, EXIT PROGRAM, dynamic-static calls use cases, more details on passing data, returning return code to the caller).

CICS stuff, hmm, those might be too specific to CICS. We may want to shift them over to the CICS chapter, depending on how deep those chapters will be.

Reply:
Sure, I will go through the chapters in the programmer Guide and rewrite the chapter